src/modules/folder/components/FolderUnlocker.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react'
+
+import { useVaultClient, VaultUnlocker } from 'cozy-keys-lib'
+
+import { isEncryptedFolder } from 'lib/encryption'
+
+const FolderUnlocker = ({ children, folder, onDismiss }) => {
+  const vaultClient = useVaultClient()
+
+  const isFolderEncrypted = isEncryptedFolder(folder)
+  const [shouldUnlock, setShouldUnlock] = useState(true)
+
+  useEffect(() => {
+    const checkLock = async () => {
+      const isLocked = await vaultClient.isLocked()
+      setShouldUnlock(isLocked)
+    }
+    if (isFolderEncrypted) {
+      checkLock()
+    }
+  }, [vaultClient, isFolderEncrypted, shouldUnlock])
+
+  if (isFolderEncrypted && shouldUnlock) {
+    return (
+      <VaultUnlocker
+        onDismiss={() => {
```

Comment:
could leverage useCallback for both,

```
  const handleDismiss = useCallback(() => {
    setShouldUnlock(false);
    onDismiss();
  }, [onDismiss]);

  const handleUnlock = useCallback(() => {
    setShouldUnlock(false);
  }, []);
  ```
  readability/perf (although mainly for readability)